File: app/src/main/java/com/mdkashif/alarm/alarm/prayer/geocoder/GetCurrentLocation.java
package com.mdkashif.alarm.alarm.prayer.geocoder;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.location.Location;
import android.location.LocationListener;
import android.location.LocationManager;
import android.os.Bundle;

import java.util.List;

import androidx.core.app.ActivityCompat;



public class GetCurrentLocation {

    private static GetCurrentLocation getCurrentLocation = new GetCurrentLocation( );

    private GetCurrentLocation() { }

    public static GetCurrentLocation getInstance( ) {
        return getCurrentLocation;
    }

    private static Location location;
    private LocationManager locationManager;

    public Location findLocation(Context con) {

        String location_context = Context.LOCATION_SERVICE;
        locationManager = (LocationManager) con.getSystemService(location_context);
        List<String> providers = locationManager.getProviders(true);
        for (String provider : providers) {
            if (ActivityCompat.checkSelfPermission(con, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(con, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            }
            locationManager.requestLocationUpdates(provider, 1000, 0,
                    new LocationListener() {

                        public void onLocationChanged(Location location) {}

                        public void onProviderDisabled(String provider) {}

                        public void onProviderEnabled(String provider) {}

                        public void onStatusChanged(String provider, int status,
                                                    Bundle extras) {}
                    });
           location = locationManager.getLastKnownLocation(provider);

        }
        return location;
    }
}

File: app/src/main/java/com/mdkashif/universalarm/alarm/prayer/geocoder/GetCurrentLocation.kt
package com.mdkashif.universalarm.alarm.prayer.geocoder

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import android.location.Location
import android.location.LocationListener
import android.location.LocationManager
import android.os.Bundle

import androidx.core.app.ActivityCompat


class GetCurrentLocation private constructor() {
    private var locationManager: LocationManager? = null

    fun findLocation(con: Context): Location? {
        locationManager = con.getSystemService(Context.LOCATION_SERVICE) as LocationManager
        val providers = locationManager!!.getProviders(true)
        for (provider in providers) {
            if (ActivityCompat.checkSelfPermission(con, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(con, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            }
            locationManager!!.requestLocationUpdates(provider, 1000, 0f,
                    object : LocationListener {

                        override fun onLocationChanged(location: Location) {}

                        override fun onProviderDisabled(provider: String) {}

                        override fun onProviderEnabled(provider: String) {}

                        override fun onStatusChanged(provider: String, status: Int,
                                                     extras: Bundle) {
                        }
                    })
            location = locationManager!!.getLastKnownLocation(provider)
        }
        return location
    }

    companion object {
        val instance = GetCurrentLocation()

        private var location: Location? = null
    }
}
